### PR TITLE
Enhance error reporting in stage.py for failed fetches

### PR DESCRIPTION
**Pull request by healther**
Part of our recent merge and upstream effort.

### CODE REVIEW

File: lib/spack/spack/stage.py [hunk 1]
```diff
@@ -490,7 +490,9 @@ def print_errors(errors):
             print_errors(errors)
 
             self.fetcher = self.default_fetcher
-            raise fs.FetchError(err_msg or 'All fetchers failed', None)
+            if not err_msg:
+                err_msg = 'All fetchers failed for {0}'.format(self.name)
+            raise fs.FetchError(err_msg, None)
```
Comment (alalazo) [inline lib/spack/spack/stage.py]:
```suggestion
            default_msg =  'All fetchers failed for {0}'.format(self.name)
            raise fs.FetchError(err_msg or default_msg, None)
```

**Review by alalazo — CHANGES_REQUESTED** [review]
A minor comment. Can you rebase to re-trigger test?

File: lib/spack/spack/stage.py (hunk 1 above)
Comment (healther) [reply lib/spack/spack/stage.py]: Done, even though I don't find this more legible. T0o much implicit function of a boolean operator (and sorry for taking so long between update bursts)

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
@healther Can you rebase this PR? Thanks!

**Comment by healther** [comment]
Done. Sorry for being out of touch for so long (unfortunately I'm not sure if I can promise you much more in the near-future)

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
No worries @healther and thanks for your contributions!

**Pull request merged**